### PR TITLE
Fix: Fix TTA and add tests

### LINKED ISSUE #157: Incorrect TTA flip transform [BUG]
**Describe the bug**
TTA transforms perform flip on -1 and -3 axis. See code ref below. -3 axis can be Z or C. We do not want to be performing the flip on C, and we potentially do not want to be performing flip on Z depending on imaging optics.

https://github.com/CAREamics/careamics/blob/eea3ecd6d4668d484a2610ac6cd8cdb38db7259f/src/careamics/transforms/tta.py#L26-L49

**Expected behavior**
Flips should happen on the YX axis only, and potentially configurable Z flips.
Maybe some more documentation in `CAREamist.predict` on why you would or would not use TTA transforms.

### PR DESCRIPTION
**Pull request by jdeschamps**
### Description

Following https://github.com/CAREamics/careamics/issues/157, this PR fixes the axes for the flipping and add tests.

- **What**: Fixing test-time augmentation used during the prediction step.
- **Why**: The axes of some of the transformations were wrong.
- **How**: Explicitly write the augmentations and add tests.

### Changes Made

- **Added**: `test_tta.py`.
- **Modified**: `CAREamist.predict` tta explanations, `ImageRestorationTTA`.

### Related Issues

- Fixes https://github.com/CAREamics/careamics/issues/157

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

File: src/careamics/transforms/tta.py [hunk 1]
```diff
@@ -1,74 +1,88 @@
...
-            rot90(x, 3, dims=(-2, -1)),
+            # original
+            input_tensor,
+            # rotations
+            rot90(input_tensor, 1, dims=axes),
+            rot90(input_tensor, 2, dims=axes),
+            rot90(input_tensor, 3, dims=axes),
+            # original flipped
+            flip(input_tensor, dims=(axes[0],)),
+            flip(input_tensor, dims=(axes[1],)),
```
Comment (melisande-c) [inline src/careamics/transforms/tta.py:45]: This doesn't actually cover every unique transformation.
There are 8 unique transformations (some combinations result in the same transformation e.g. flip X followed by flip Y is equivalent to rotate 180).
To cover every possible combination, do all the rotations and then the combination of each rotation with a flip.
```python
augmented = [
   # original
   input_tensor,
   # rotations
   rot90(input_tensor, 1, dims=axes),
   rot90(input_tensor, 2, dims=axes),
   rot90(input_tensor, 3, dims=axes),
   # rotations + flip
   flip(input_tensor, dims=(-1, )),
   flip(rot90(input_tensor, 1, dims=axes), dims=(-1, )),
   flip(rot90(input_tensor, 2, dims=axes), dims=(-1, )),
   flip(rot90(input_tensor, 3, dims=axes), dims=(-1, )),
]
```

Comment (jdeschamps) [reply src/careamics/transforms/tta.py:45]: Correct, but it is done in the lines that follow 😃 
https://github.com/CAREamics/careamics/blob/cfe269fcdc72d02e2f65490ad656262d5ab79692/src/careamics/transforms/tta.py#L49

```python
        augmented.extend(
            [
                flip(augmented[1], dims=(axes[0],)),
                flip(augmented[1], dims=(axes[1],)),
            ]
        )
```

I didn't want to recompute the rotations again (which might not be computationally faster...).

Comment (melisande-c) [reply src/careamics/transforms/tta.py:45]: My mistake! I guess I read it too quickly

> - **How**: Explicitly write the augmentations

I partly assumed this from your description in the PR.

**Review by melisande-c — APPROVED** [review]

File: src/careamics/transforms/tta.py (hunk 1 above)
Comment (jdeschamps) [reply src/careamics/transforms/tta.py:45]: Ahah ouch, point taken. 😄

**Pull request merged**